### PR TITLE
Add transport status to subchannel picked log

### PR DESCRIPTION
**Pull request by JamesNK**
Help debug call issues by reporting information about the subchannel transport status when a subchannel is picked.

### CODE REVIEW

File: src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs [hunk 1]
```diff
@@ -82,6 +82,24 @@ internal class SocketConnectivitySubchannelTransport : ISubchannelTransport, IDi
     private object Lock => _subchannel.Lock;
     public BalancerAddress? CurrentAddress => _currentAddress;
     public TimeSpan? ConnectTimeout { get; }
+    public TransportStatus TransportStatus
+    {
+        get
+        {
+            lock (Lock)
+            {
+                if (_initialSocket != null)
```
Comment (amcasey) [inline src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs:91]: Without knowing anything about this type, I would have guessed that having active streams was more specific than having an initial(ized?) socket and, therefore, would have assumed that that status would take precedence.

File: src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs [hunk 2]
```diff
@@ -82,6 +82,24 @@ internal class SocketConnectivitySubchannelTransport : ISubchannelTransport, IDi
...
+                {
+                    return TransportStatus.InitialSocket;
+                }
+                if (_activeStreams.Count > 0)
+                {
+                    return TransportStatus.ActiveStream;
+                }
+                return TransportStatus.Unknown;
```
Comment (amcasey) [inline src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs:99]: Is it actually unknown or does this mean uninitialized?

**Review by amcasey — APPROVED** [review]
The meanings of the enum values aren't very clear to an outsider.

File: src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs (hunk 1 above)
Comment (JamesNK) [reply src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs:91]: Both will never be true. Order doesn't matter.

File: src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs (hunk 2 above)
Comment (JamesNK) [reply src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs:99]: I renamed it to `NotConnected`. There are a bunch of reasons to be in that state. `NotConnected` name fits them all.

Comment (amcasey) [reply src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs]: Much clearer.  Thanks!

**Pull request merged**